packages/bits/src/lib/sorter/sorter-key-control.service.ts

```diff
@@ -0,0 +1,100 @@
+import { Injectable, QueryList } from "@angular/core";
+import { ListKeyManager, LiveAnnouncer } from "@angular/cdk/a11y";
+import isNull from "lodash/isNull";
+
+import { MenuItemBaseComponent } from "../menu";
+import { OverlayComponent } from "../overlay/overlay-component/overlay.component";
+import { KEYBOARD_CODE } from "../../constants";
+import { ANNOUNCER_CLOSE_SORTER_LIST_MESSAGE, ANNOUNCER_OPEN_SORTER_LIST_MESSAGE_SUFFIX } from "./constants";
+
+@Injectable()
+export class SorterKeyControlService {
+    public overlay: OverlayComponent;
+    public menuItems: QueryList<MenuItemBaseComponent>;
+    private keyboardEventsManager: ListKeyManager<MenuItemBaseComponent>;
+
+    constructor(private liveAnnouncer: LiveAnnouncer) {}
+
+    public initKeyboardManager(): void {
+        this.keyboardEventsManager = new ListKeyManager(this.menuItems).withVerticalOrientation();
+        // TODO Uncomment in the scope of NUI-
+        // this.keyboardEventsManager.setFirstItemActive();
+
+        // TODO Remove this in the scope of NUI-
+        this.keyboardEventsManager.setActiveItem(-1);
+    }
+
+    public handleKeydown(event: KeyboardEvent): void {
+        this.overlay.showing ? this.handleOpenKeyDown(event) : this.handleClosedKeyDown(event);
+    }
+
+    public getActiveItemIndex(): number | null {
+        return this.keyboardEventsManager.activeItemIndex;
+    }
+
+    public announceDropdown(open: boolean): void {
+        const menuListLength = this.menuItems.toArray().length;
+        const message = open ? `${menuListLength} ${ANNOUNCER_OPEN_SORTER_LIST_MESSAGE_SUFFIX}` : ANNOUNCER_CLOSE_SORTER_LIST_MESSAGE;
+        this.liveAnnouncer.announce(message);
+    }
```

Comment:
I feel like we can do without the parameter since we have a reference to the overlay:

```suggestion
    public announceDropdown(): void {
        const menuListLength = this.menuItems.toArray().length;
        const message = this.overlay.showing ? `${menuListLength} ${ANNOUNCER_OPEN_SORTER_LIST_MESSAGE_SUFFIX}` : ANNOUNCER_CLOSE_SORTER_LIST_MESSAGE;
        this.liveAnnouncer.announce(message);
    }
```